community/kernel/src/main/java/org/neo4j/kernel/IdType.java

```diff
@@ -19,49 +19,83 @@
  */
 package org.neo4j.kernel;
 
-import org.neo4j.unsafe.impl.internal.dragons.FeatureToggles;
+import java.util.Arrays;
+import java.util.List;
 
 /**
  * @deprecated This will be moved to internal packages in the next major release.
  */
 @Deprecated
-public enum IdType
+public class IdType
```

Comment:
To my understanding it seems that this, rather big, change w/ regards to `IdType` and all code this affects has to do with the fact that its state is now mutable, i.e. `allowAggressiveIdReuse`. I'd prefer it if we would instead separate `IdType` from its mutable state and keep IdType what it is. That has to be simpler, hasn't it?